In [4]:
import numpy as np

## Задача 1

### Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).

В нашем случае Х - число выпадения правильных ответов

n - число испытаний. В нашем случае их 5

p - вероятность появления события. В нашем случае она равна 1/4 для каждого испытания

Соответственно, q = 1 - 1/4 = 3/4

In [5]:
# 𝑀(𝑋)=𝑛𝑝

m = 5 * 1/4
print(m)

1.25


In [6]:
# 𝐷(𝑋)=𝑛𝑝𝑞

d = 5 * 1/4 * 3/4
print(d)

0.9375


Закон распределения числа правильных ответов при простом угадывании будет выглядеть следующим образом:

Вводные данные все те же - n=5, p=1/4, q=3/4

In [44]:
def bernulli(k): 
    P = (np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k))) * (p**k) * (q**(n-k))
    return P

In [30]:
n = 5
p = 1/4
q = 3/4

In [31]:
bernulli(0)

0.2373046875

In [32]:
bernulli(1)

0.3955078125

In [17]:
bernulli(2)

0.263671875

In [18]:
bernulli(3)

0.087890625

In [19]:
bernulli(4)

0.0146484375

In [20]:
bernulli(5)

0.0009765625

Итак, закон распределения такой случайной величины:

<table border="3">
    <tr>
        <th>$x$</th>
        <td>0</td>
        <td>1</td>
        <td>2</td>
        <td>3</td>
        <td>4</td>
        <td>5</td>
    </tr>
    <tr>
        <th>$P(X = x)$</th>
        <td>0.237</td>
        <td>0.396</td>
        <td>0.234</td>
        <td>0.088</td>
        <td>0.015</td>
        <td>0.001</td>
    </tr>
</table>

## Задача 2

### Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?

p - вероятность попадания в цель при одном выстреле, в нашем случае 0.01

n - количество выстрелов, в нашем случае 200

k - количество вероятных попаданий в цель, в нашем случае это диапазон 5 =< k <= 10

Как разобрали на уроке, используем биноминальное распределение, т.к. нет временной привязки и ограничения.

In [35]:
n = 200
p = 0.01
q = 1-p

In [37]:
sum(bernulli(k) for k in range(5, 11))

0.051739381860374606

Таким образом, искомая вероятность равна 0.052

## Задача 3

### В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?

#### Подсказка. Как посчитать P(X >= k) для величины X, имеющей распределение Пуассона? Здесь предстоит немножко покодить.

p - вероятность попадания в цель при одном выстреле, в нашем случае 0.01

pk - вероятность, к которой нам нужно прйти после серии сипытаний, в нашем случае 0.9

k - количество вероятных попаданий в цель, в нашем случае это диапазон k >= 10 (хотя бы 10 раз - это значит 10 и более). Поскольку диапазон не ограничет, то легче будет найти обратную вероятность для 0 < k < 10 и вычесть ее из единицы.

In [65]:
def bn (n, k, p): 
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k)) * (p**k) * ((1 - p)**(n-k))

In [66]:
def bn_ob(n, k, p):
    return 1 - sum(bn(n, i, p) for i in range (k))

In [67]:
p = 0.01
k = 10
n = 10
pk = 0

while pk < 0.9:
    n += 1
    pk = bn_ob(n, k, p)
    
print(n, pk)

1418 0.9000040527998662
